go/src/socialapi/workers/payment/subscription.go

```diff
@@ -121,20 +121,20 @@ func EnsureSubscriptionForGroup(groupName string, params *stripe.SubParams) (*st
 	}
 
 	now := time.Now().UTC()
-	thirtyDaysLater := now.Add(30 * 24 * time.Hour).Unix()
-	sevenDaysLater := now.Add(7 * 24 * time.Hour).Unix()
+	// apply plan's default trial period
+	if plan := GetPlan(params.Plan); plan != nil && plan.TrialPeriod != 0 {
+		params.TrialEnd = now.Add(time.Duration(plan.TrialPeriod) * 24 * time.Hour).Unix()
+	}
 
-	if params.TrialEnd != 0 {
-		// we only allow 0, 7 and 30 day trials
-		if params.TrialEnd < sevenDaysLater {
-			params.TrialEnd = sevenDaysLater
-		}
+	// if group is not created within last ten minutes, subtract its duration from the trial end date
+	groupCreatedAt := group.Id.Time().UTC()
 
-		if params.TrialEnd > sevenDaysLater {
-			params.TrialEnd = thirtyDaysLater
-		}
+	if params.TrialEnd != 0 && !groupCreatedAt.Add(time.Minute*5).After(now) {
```

Comment:
yep, even 5 mins is a big duration, but i dont have any scientific way to measure this, i just guestimated some big random duration